services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/LegAnnotation.java

```diff
@@ -219,6 +223,28 @@ public abstract static class Builder extends DirectionsJsonObject.Builder<Builde
     @NonNull
     public abstract Builder trafficTendency(@Nullable List<Integer> trafficTendency);
 
+    /**
+     * Use this method to add parameters which are not present in the model yet but are supported
+     * on the Directions API side in the response.
+     * Use it for experimental parameters.
+     *
+     * @param unrecognizedProperties parameters to add to request
+     */
+    @NonNull
+    public Builder unrecognizedProperties(
+      @Nullable Map<String, JsonElement> unrecognizedProperties
```

Comment:
`RouteOptions.Builder` exposes this map as `Map<String, String>`, we should align this one way or the other. For the general purpose `JsonElement` does look like a better candidate, but again only if we don't want to expose the `SerializableJsonElement`. @VysotskiVadim I remember you had some thoughts about it in the past.

Reply:
Yes, here we definitely need `JsonElement`. The structure will be complex.
For `RouteOptions` it's only about get parameters, that's why strings are used.
These are different use cases so I'm not sure if we should complicate `RouteOptions` usage with converting it all to `JsonElement`. But we could have 2 methods for `RouteOptions` to keep both consistency and usability.

Reply:
>  @VysotskiVadim I remember you had some thoughts about it in the past.

the initial idea was to expose as little as possible to simplify future changes. 

> For RouteOptions it's only about get parameters, that's why strings are used.

that's right.

> These are different use cases so I'm not sure if we should complicate RouteOptions usage with converting it all to JsonElement. 

I agree with @dzinad. I don't see value in aligning unrecognised properties API between request and response models.

Reply:
I don't fully agree. Whether we look at request or response object, they are all `DirectionsJsonObject`s (or refresh) and support the unrecognized fields in exactly the same way, and that's why I think they should also expose the same API to mutate or reset the properties. 

We started with `RouteOptions` needing some form of unrecognized properties setter, now we need to add the same to `LegAnnotation` and this could be true at any point in time for any `DirectionsJsonObject`, depending on what's `experimental` in the Directions API at a certain time.

Couldn't we expose `Map<String, JsonElement>` getter and setters for all `DirectionsJsonObject`s?

Reply:
We can if we leave the convenience method for `RouteOptions`.

Reply:
Why couldn't we use only `Map<String, JsonElement>` for `RouteOptions` as well?

The only change developers would have to make is wrap their `string` values in `JsonPrimitive(value)`. That's actually even more flexible because other than `string` parameters are also acceptable by the Directions API. Of course we can leave the existing function in place but I'd rather deprecate it.

Reply:
1) It's a breaking change and that's something that would definitely affect the users;
2) In the majority of the cases users only need `Map<String, String>`. Making them use `Map<String, JsonElement>` will force them to write more boiler-plate code on their side. 

I'm strongly in favour of leaving the convenience method as-is. We'll provide a more flexible options: if they want something other than strings, they can use the new method. And I'm not suggesting to add `Map<String, String>` everywhere: only leave it where it already is and where it makes sense to have it.

Reply:
> It's a breaking change and that's something that would definitely affect the users;

We'd not be removing the old function, only deprecating it.

> In the majority of the cases users only need Map<String, String>. Making them use Map<String, JsonElement> will force them to write more boiler-plate code on their side.

`Map<String, String>` is limiting and actually somewhat incorrect. For example, look at `ev_max_charge` parameter in https://docs.mapbox.com/api/navigation/directions/#parameters-for-electric-vehicle-routing - it's an integer but we'd recommend (and today require) developers pass it as a `string`. `JsonPrimitive(integer)` would be more accurate here.

In any case, we can discuss and consider the deprecation separately, the old function has to stay in place regardless.